### PR TITLE
Fix #552 - making email field required

### LINKED ISSUE #552: Frontend: Add validation before sending invite to other team members
- [ ] Handle errors (email required & email valid ) with ng-messages as it is done in auth pages.
- [x] Show notification on invitation success after closing the modal of email invite (depend on issue #482 )

### PR DESCRIPTION
**Pull request by Obtin**
Fix #552

### CODE REVIEW

**Comment by Obtin** [comment]
@RishabhJain2018 working on that will be completed soon. :-)

**Comment by RishabhJain2018** [comment]
@Obtin Please let us know when this is complete :-)

**Comment by Obtin** [comment]
invite1
invite2
invite3

**Comment by spyshiv** [comment]
Thanks @Obtin for solving the issue that was pending for a long time. Can you make popup UI better? It needs better alignment of texts and input box.

**Comment by Obtin** [comment]
@spyshiv, Please check if any other changes are suggested. Should I wait to make PR for #1503
before this is merged.Thank you.

**Comment by spyshiv** [comment]
@Obtin I have already reviewed this PR. Please do the changes so that I can merge this PR.

**Comment by Obtin** [comment]
emailinvite1
@spyshiv 
emailinvite2

**Comment by spyshiv** [comment]
@Obtin The UI part doesn't look up to mark.
@aka-jain Need your thoughts here regarding design of it.

**Comment by Obtin** [comment]
@spyshiv , Does this look good.
email2345

**Comment by aka-jain** [comment]
@Obtin please take reference from other modal design.
screen shot 2018-02-04 at 19 51 10

**Comment by Obtin** [comment]
@aka-jain @spyshiv, Please have a look
invitebyemail.

File: frontend/src/css/modules/base.scss [hunk 1]
```diff
@@ -180,8 +180,16 @@ span.inline {
 
 /*main intro page*/
 
-.ev-md-container {
-    padding: 40px;
+.ev-md-dialog-content {
```
Comment (aka-jain) [inline frontend/src/css/modules/base.scss]: @Obtin why you have deleted `ev-md-container`?

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 2]
```diff
@@ -309,17 +309,44 @@
...
-                .title('Invite others to this Team')
-                .textContent('Enter the email address of the person')
-                .placeholder('email')
-                .ariaLabel('')
-                .targetEvent(ev)
-                .ok('Send Invite')
-                .cancel('Cancel');
 
-            $mdDialog.show(confirm).then(function(result) {
+            $mdDialog.show({
```
Comment (aka-jain) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please create a separate variable for this template like we did earlier with `confirm` variable

File: frontend/src/js/controllers/teamsCtrl.js [hunk 3]
```diff
@@ -326,17 +326,45 @@
...
-                .textContent('Enter the email address of the person')
-                .placeholder('email')
-                .ariaLabel('')
-                .targetEvent(ev)
-                .ok('Send Invite')
-                .cancel('Cancel');
 
-            $mdDialog.show(confirm).then(function(result) {
+            $mdDialog.show({
+                 clickOutsideToClose: true,
```
Comment (aka-jain) [inline frontend/src/js/controllers/teamsCtrl.js]: Same as above.

**Comment by deshraj** [comment]
Glad to see that we are making progress on this. Thanks @Obtin for your patience and contribution. :)

File: frontend/src/css/modules/base.scss (hunk 1 above)
Comment (Obtin) [reply frontend/src/css/modules/base.scss]: this is because I changed the `ev-md-container` to 'ev-md-content' in template

**Comment by Obtin** [comment]
@deshraj @aka-jain Please review it and let me know if it needs any other changes too. Thank You.

**Comment by spyshiv** [comment]
@Obtin FYI

Current UI
screenshot from 2018-02-21 12 36 16

Expected UI
screenshot from 2018-02-21 12 39 10

Please do necessary changes to make this UI perfect. Let me know if you face any issue

**Comment by Obtin** [comment]
@spyshiv, Please have a look at it.
fix552

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 4]
```diff
@@ -309,17 +309,44 @@
...
-                .title('Invite others to this Team')
-                .textContent('Enter the email address of the person')
-                .placeholder('email')
-                .ariaLabel('')
-                .targetEvent(ev)
-                .ok('Send Invite')
-                .cancel('Cancel');
-
-            $mdDialog.show(confirm).then(function(result) {
+            var inviteTemplate =
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: @Obtin I think this isn't the right way of adding a template. We must use a separate file for it and then we can give it's path here. Please let me know if something is unclear.

File: frontend/src/js/controllers/teamsCtrl.js [hunk 5]
```diff
@@ -324,19 +324,48 @@
         };
 
 
+        var inviteTemplate =
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/teamsCtrl.js]: @Obtin Same as above!

**Comment by Obtin** [comment]
closing this pr due to certain git issues. I have created another pull requested with the desired changes.
#1542

**Pull request closed without merging**